src/sif/recost.cc

```diff
@@ -72,17 +72,21 @@ void recost_forward(baldr::GraphReader& reader,
       throw std::runtime_error("Node cannot be found");
     }
 
-    // this node is not allowed
-    if (node && !costing.Allowed(node)) {
-      throw std::runtime_error("This path requires different node access than this costing allows");
-    }
-
     // grab the edge
     edge = reader.directededge(edge_id, tile);
     if (!edge) {
       throw std::runtime_error("Edge cannot be found");
     }
 
+    // re-derive uturns, would have been nice to return this but we dont know the next edge yet
+    if (label.opp_local_idx() == edge->localedgeidx())
```

Comment:
why add a branch to this code path, we simply store the value of this comparison?

Reply:
because by default we use `deadend` flag from the edge

Reply:
hm. from the other hand, if it's deadend, this condition will be always true